src/recvmsg.c

```diff
@@ -32,14 +32,14 @@ int __doprocess(lua_State *L)
    if (perr)
       return lua_error(L);
 
-   return 0;
+   return 1;//lua_toboolean(L, -1);
 }
 
 int ulp_recvmsg(struct sock *sk, struct msghdr *msg, size_t len,
       int nonblock, int flags, int *addr_len)
 {
    struct tcp_sock *tp = tcp_sk(sk);
-   lua_State *L = sk_ulp_data(sk)->L;
+   lua_State *L = sk_conn_ulp_data(sk)->L;
```

Comment:
should be still `sk_ulp_data()` we can differentiate the sockets using the socket state